OptimoveSDK/Sources/Classes/Optimobile/InApp/InAppPresenter.swift

```diff
@@ -318,29 +341,23 @@ class InAppPresenter : NSObject, WKScriptMessageHandler, WKNavigationDelegate{
         let type = body["type"] as! String
         
         if (type == "READY") {
-              messageQueueLock.wait()
-              defer {
-                  messageQueueLock.signal()
-              }
```

Comment:
This lock is now acquired & released in presentFromQueue directly to save callers needing to reason about upcoming queue operations